src/transformers/models/bert/modeling_bert.py

```diff
@@ -1460,15 +1461,16 @@ def forward(
     BERT_START_DOCSTRING,
 )
 class BertForSequenceClassification(BertPreTrainedModel):
-    def __init__(self, config):
+    def __init__(self, config, init_weights=True):
```

Comment:
This is going to and up adding a lot of code everywhere. I agree we can't attach `init_weights` to the config but it could be done by reading the `init_weights` of some internal variable of transformers, and we could design a context manager:
```
with no_init_weights():
    model = cls(config)
```
that would switch the variable on and off. That would be clean while avoiding a lot of code change on our side, what do you think?

Reply:
Great idea - I like it a lot! 

We could then simply make `self.init_weights()` not do anything if the model call: `model = cls(config)` is wrapped into a context manager -> I like it!

Reply:
I like this approach!

It's a bit similar to what I suggested in my version of this PR, albeit I was just thinking to add and immediately then remove `config.foo` attribute - poor man's context. But I think my idea wasn't well thought through because I noticed `init_weights` could be called more than once. based on the log I added.